src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -0,0 +1,98 @@
+package assisted_installer_controller
+
+import (
+	"context"
+	"time"
+
+	"github.com/eranco74/assisted-installer/src/inventory_client"
+	"github.com/eranco74/assisted-installer/src/k8s_client"
+	"github.com/eranco74/assisted-installer/src/ops"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	done = "Done"
+)
+
+type ControllerConfig struct {
+	ClusterID string `envconfig:"CLUSTER_ID" required:"true" `
+	Host      string `envconfig:"INVENTORY_HOST" required:"true"`
+	Port      int    `envconfig:"INVENTORY_PORT" required:"true"`
+}
+
+type Controller interface {
+	WaitAndUpdateNodesStatus()
+}
+
+type controller struct {
+	ControllerConfig
+	log *logrus.Logger
+	ops ops.Ops
+	ic  inventory_client.InventoryClient
+	kc  k8s_client.K8SClient
+}
+
+func NewController(log *logrus.Logger, cfg ControllerConfig, ops ops.Ops, ic inventory_client.InventoryClient, kc k8s_client.K8SClient) *controller {
+	return &controller{
+		log:              log,
+		ControllerConfig: cfg,
+		ops:              ops,
+		ic:               ic,
+		kc:               kc,
+	}
+}
+
+func (c *controller) WaitAndUpdateNodesStatus() {
+	c.log.Infof("Waiting till all nodes will join and update status to assisted installer")
+	apiContext, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	nodeNameAndCondition := make(map[string]string)
+	wait.Until(func() {
+		assistedInstallerNodes, nodes, err := c.getNodes()
+		if err != nil {
+			return
+		}
+		for _, node := range nodes.Items {
+			_, ok := nodeNameAndCondition[node.Name]
+			if ok {
+				return
+			}
+			c.log.Infof("Found new joined node %s with id %s", node.Name, node.Status.NodeInfo.SystemUUID)
+			err := c.updateHostStatus(node.Status.NodeInfo.SystemUUID, done)
+			if err == nil {
+				nodeNameAndCondition[node.Name] = node.Status.NodeInfo.SystemUUID //node.Status.Conditions
+			}
+
+		}
+		c.log.Infof("Found %d nodes: %+v", len(nodes.Items), nodeNameAndCondition)
+		if len(nodes.Items) >= len(assistedInstallerNodes) {
+			c.log.Infof("WaitAndUpdateNodesStatus - Done")
+			cancel()
+		}
+	}, 30*time.Second, apiContext.Done())
+}
+
+func (c *controller) updateHostStatus(hostId string, newStatus string) error {
```

Comment:
You don't really need this method, you get the same logging in `c.ic.UpdateHostStatus`